Hitachi344: Add Swing(H) and improve Swing(V)

crankyoldgit opened the pull request:
* Correct Swing(V) control
* Add Swing(H) support.
* Update common A/C support.
* Reduce duplicate code between 344 & 424 classes.
* add & update supporting Unit Tests.
* Fix typo

Fixes #1134

crankyoldgit committed:
Hitachi344: Add Swing(H) and improve Swing(V)
* Correct Swing(V) control
* Add Swing(H) support.
* Update common A/C support.
* Reduce duplicate code between 344 & 424 classes.
* add & update supporting Unit Tests.
* Fix typo

Fixes #1134

NiKiZe commented:
Will check this more when my head is less mushy, but a quick check didn't see anything.

crankyoldgit committed:
Update supported models

NiKiZe commented on src/ir_Hitachi.h:
```diff
@@ -11,7 +11,8 @@
 //   Brand: Hitachi,  Model: R-LT0541-HTA/Y.K.1.1-1 V2.3 remote (HITACHI_AC1)
-//   Brand: Hitachi,  Model: RAS-22NK remote (HITACHI_AC344)
+//   Brand: Hitachi,  Model: RAS-22NK A/C (HITACHI_AC344)
+//   Brand: Hitachi,  Model: RF11T1 remote (HITACHI_AC344)
```
Not an issue, just a question
There is some data about support in the `.cpp` file as well,
Should that data be removed/moved, and the support data should only be in the `.h` file?

Also seen some support data that does not match the new format, want to change that? or are they left as is because support are unsure?
Should `SupportedProtocols.md` be regenerated?

crankyoldgit replied:
IIRC, the tool scrapes both *.h & *.cpp files to build up it's picture.
The _general_ rule of thumb is "store it in the .h file if it exists.".

> Should that data be removed/moved, and the support data should only be in the .h file?

Probably. Feel free to.

> Also seen some support data that does not match the new format, want to change that? or are they left as is because support are unsure?

Probably. Feel free to. I did a quick pass over the code when we (@sillyfrog & I) added this sort of tool. I probably missed things.

> Should SupportedProtocols.md be regenerated?

I was going with your suggestion of a while back, of really only running at when we do a release. There are good reasons for and against running it out-of-release-cycle. I'm now on the fence after your discussion last time. ;-)

NiKiZe replied:
So I would then like to propose (and will try to implement):
* if there is a .h file, warn if there is anything in the cpp file
* try to search for old format data, and warn about it

Good on the generation part, just want to be consistent.

crankyoldgit replied:
SGTM

crankyoldgit committed:
Improve doxygen comments.

crankyoldgit committed:
Merge branch 'Hitachi344Swing' of https://github.com/crankyoldgit/IRremoteESP8266 into Hitachi344Swing

crankyoldgit commented:
Note to self: Wait a few days before merging to allow user feedback.

crankyoldgit committed:
Hitachi344: Attempt to fix swingv operation.
For #1134

crankyoldgit committed:
Fix linter issue.
`test/ir_Hitachi_test.cpp:1968:  Tab found; better to use spaces  [whitespace/tab] [1]`
* Mutter-mutter-cut-n-paste-mutter-mutter-swearword!

crankyoldgit committed:
Another attempt at fixing swingv
For #1134

crankyoldgit committed:
Attempt 3: Use the same method as Hitachi424
For #1134

NiKiZe commented on src/ir_Hitachi.cpp:
```diff
@@ -1366,10 +1374,93 @@ void IRHitachiAc344::setRaw(const uint8_t new_code[], const uint16_t length) {
+  if (position > kHitachiAc344SwingHLeftMax)
+    return setSwingH(kHitachiAc344SwingHMiddle);
+  setBits(&remote_state[kHitachiAc344SwingHByte], kHitachiAc344SwingHOffset,
+         kHitachiAc344SwingHSize, position);
```
indent one more space?

crankyoldgit replied:
fixed.

NiKiZe commented on src/ir_Hitachi.cpp:
```diff
@@ -1366,10 +1374,93 @@ void IRHitachiAc344::setRaw(const uint8_t new_code[], const uint16_t length) {
+/// @return The current position horizontal swing is set to.
+uint8_t IRHitachiAc344::getSwingH(void) {
+  return GETBITS8(remote_state[kHitachiAc344SwingHByte],
+                 kHitachiAc344SwingHOffset, kHitachiAc344SwingHSize);
```
indent one more space?

crankyoldgit replied:
Fixed.

crankyoldgit commented:
Confirmed working in https://github.com/crankyoldgit/IRremoteESP8266/issues/1134#issuecomment-636421629

As the code has evolved a lot since you last reviewed it @NiKiZe I'm requesting a re-review.

crankyoldgit committed:
Fixes from code review.

NiKiZe commented:
> Confirmed working in [#1134 (comment)](https://github.com/crankyoldgit/IRremoteESP8266/issues/1134#issuecomment-636421629)
> 
> As the code has evolved a lot since you last reviewed it @NiKiZe I'm requesting a re-review.

Was expecting it, and didn't do much while it wasn't confirmed working. :D
